qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/define_inputs.py

```diff
@@ -0,0 +1,313 @@
+    number_histograms = len(histogram_indexing)
+if len(x_min) != 0 and len(x_min) != number_histograms:
+    raise ValueError(f"The number of x_min values ({len(x_min)}) must equal the"
+                     + f"number of histograms ({number_histograms}))")
+
+if histogram_indexing and len(data_files) > 1:
+    raise ValueError(f"Histogram indexing is currently only supported, when the "
+                     + f"number of data_files ({len(data_files)}) == 1")
+
+if refinement_method == 'Pawley' and not mantid_pawley_reflections:
+    raise ValueError(f"No Pawley Reflections were generated for the phases provided. Not calling GSASII.")
+
+
+'''exec'''
+
+gsas2_inputs = parse_inputs.Gsas2Inputs(
+    path_to_gsas2=path_to_gsas2,
+    temporary_save_directory=temporary_save_directory,
+    data_directory=data_directory,
+    project_name=project_name,
+    refinement_method=refinement_method,
+    refine_background=refine_background,
+    refine_microstrain=refine_microstrain,
+    refine_sigma_one=refine_sigma_one,
+    refine_gamma=refine_gamma,
+    refine_histogram_scale_factor=refine_histogram_scale_factor,
+    data_files=data_files,
+    histogram_indexing=histogram_indexing,
+    phase_files=phase_files,
+    instrument_files=instrument_files,
+    limits=[x_min, x_max],
+    mantid_pawley_reflections=mantid_pawley_reflections,
+    override_cell_lengths=override_cell_lengths
+)
+
+call_gsas2 = (path_to_gsas2 + "bin/python "
+              + "/home/danielmurphy/mantid/qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/"
+              + "engineering_diffraction/tabs/gsas2/call_G2sc.py "
+              + parse_inputs.convert_Gsas2Inputs_to_json(gsas2_inputs)
+              )
+
+start = time.time()
+out_call_gsas2, err_call_gsas2 = call_subprocess(call_gsas2)
+gsas_runtime = time.time() - start
+print(format_shell_output(title="Commandline output from GSAS-II", shell_output_string=out_call_gsas2))
+
+
+gsas_project_filename = project_name + '.gpx'
+gsas_project_filepath = os.path.join(temporary_save_directory, gsas_project_filename)
+if gsas_project_filename not in os.listdir(temporary_save_directory):
+    raise FileNotFoundError("GSAS-II call must have failed, as the output project file was not found.",
+                            format_shell_output(title="Errors from GSAS-II", shell_output_string=err_call_gsas2))
+
+gsas_log_filename = project_name + '.lst'
+gsas_log_filepath = os.path.join(temporary_save_directory, gsas_log_filename)
+if gsas_log_filename not in os.listdir(temporary_save_directory):
+    raise FileNotFoundError("GSAS-II call must have failed, as the output log file was not found.",
+                            format_shell_output(title="Errors from GSAS-II", shell_output_string=err_call_gsas2))
+
+print(f"\nGSAS-II call complete in {gsas_runtime} seconds.\n")
+print(f"GSAS-II .lst result file found. Opening {gsas_log_filename}")
+read_gsas_lst_and_print_wR(gsas_log_filepath)
```

Comment:
Use mantid logging instead of printing

Reply:
Cannot pause the process with logger.error so still raise Python Errors normally